### PR TITLE
Setup Vale linter as a GHA workflow

### PR DESCRIPTION
**Pull request by ankatiyar**
## Description
Resolve https://github.com/kedro-org/kedro/issues/2518

## Development notes
See PR - https://github.com/kedro-org/kedro/pull/2954 to see this workflow in action
* I've extended the Spellings test from Vale to ignore certain words in `.github/styles/Kedro/ignore.txt` (this is a bunch of words that came up as errors when I ran vale locally) and `.github/styles/Kedro/ignore-names.txt` (proper nouns/names of maintainers). This might get annoying to maintain.
* This workflow only leaves suggestions as review comments on PRs for changed lines. These comments can then be incorporated or ignored (marked as Resolved?)
* Haven't added any other style guidelines. I'll leave that for @stichbury to decide.
    * These can be added as `yaml` files under `.github/styles/Kedro`
    * See https://github.com/DataDog/datadog-vale/tree/main/styles/Datadog for inspiration

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: .github/styles/Kedro/ignore.txt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+Kedro
+kedro
+Kedro's
+kedro's
```
Comment (astrojuanlu) [inline .github/styles/Kedro/ignore.txt:4]: Do we want to favour a specific capitalization?

Comment (stichbury) [inline .github/styles/Kedro/ignore.txt:4]:
```suggestion
Kedro
Kedro's
```

**Review by SajidAlamQB — APPROVED** [review]
This is awesome! I really like this idea, thank you @ankatiyar

**Review by stichbury — APPROVED** [review]
Happy to approve this, it's fantastic. Thanks! 🌟

**Comment by DimedS** [comment]
Cool and helpful tool, thanks @ankatiyar 👍

**Pull request merged**